catalog/transformer/catalog-transformer-geojson-input/src/main/java/ddf/catalog/transformer/input/geojson/GeoJsonInputTransformer.java

```diff
@@ -190,11 +196,38 @@ private void validateTypeValue(Map<String, Object> rootObject)
     return properties;
   }
 
-  private MetacardImpl getMetacard(String propertyTypeName) throws CatalogTransformerException {
+  private MetacardImpl getMetacard(String propertyTypeName, Map<String, Object> properties)
+      throws CatalogTransformerException {
     if (isEmpty(propertyTypeName) || metacardTypes == null) {
       LOGGER.debug(
-          "MetacardType specified in input is null or empty.  Assuming default MetacardType");
-      return new MetacardImpl();
+          "MetacardType specified in input is null or empty.  Trying all transformers in order...");
+      Optional<MetacardImpl> first =
+          inputTransformers == null
+              ? Optional.of(new MetacardImpl())
+              : inputTransformers
+                  .stream()
+                  .map(
+                      serviceReference -> {
+                        try {
+                          InputTransformer inputTransformer =
+                              (InputTransformer)
+                                  FrameworkUtil.getBundle(this.getClass())
+                                      .getBundleContext()
+                                      .getService(serviceReference);
+                          return new MetacardImpl(
+                              inputTransformer.transform(
+                                  new ByteArrayInputStream(
+                                      ((String) properties.get("metadata")).getBytes())));
+                        } catch (Exception e) {
+                          LOGGER.debug(
+                              "Error calling transformer: " + serviceReference.getProperty("id"),
+                              e);
+                        }
+                        return null;
+                      })
+                  .filter(Objects::nonNull)
+                  .findFirst();
+      return first.orElse(new MetacardImpl());
```

Comment:
Do you need this `orElse(new MetacardImpl()` in addition to the `Optional.of(new MetacardImpl())` line above?

Reply:
I think so. The optional guards against a null list of transformers, while the else is a fall-through if no transformer can handle the metadata.